pkg/client/standalone.go

```diff
@@ -74,10 +80,61 @@ func (c *StandaloneClient) Ingest(ctx context.Context, datasetID string, name st
 	return ids, err
 }
 
-func (c *StandaloneClient) IngestPaths(ctx context.Context, datasetID string, opts *IngestPathsOpts, paths ...string) (int, error) {
+func (c *StandaloneClient) IngestFromWorkspace(ctx context.Context, datasetID string, opts *IngestWorkspaceOpts, file string) error {
+	_, err := getOrCreateDataset(ctx, c, datasetID, true)
+	if err != nil {
+		return err
+	}
+
+	file = strings.TrimPrefix(file, "ws://")
+
+	meta := make(map[string]any, len(opts.Metadata))
+	for k, v := range opts.Metadata {
+		meta[k] = v
+	}
+
+	finfo, err := c.GPTScript.StatFileInWorkspace(ctx, file)
+	if err != nil {
+		return fmt.Errorf("failed to stat file %q: %w", file, err)
+	}
+
+	fileContent, err := c.GPTScript.ReadFileInWorkspace(ctx, file)
+	if err != nil {
+		return fmt.Errorf("failed to read file %q: %w", file, err)
+	}
+
+	iopts := datastore.IngestOpts{
+		FileMetadata: &index.FileMetadata{
+			Name:         finfo.Name,
+			AbsolutePath: fmt.Sprintf("ws://%s/%s", finfo.WorkspaceID, file),
+			Size:         finfo.Size,
+			ModifiedAt:   finfo.ModTime,
+		},
+		IsDuplicateFuncName: opts.IsDuplicateFuncName,
+		ExtraMetadata:       meta,
+		TextSplitterOpts:    opts.TextSplitterOpts,
+		IngestionFlows:      opts.IngestionFlows,
+	}
+
+	_, err = c.Ingest(log.ToCtx(ctx, log.FromCtx(ctx).With("filepath", file).With("absolute_path", iopts.FileMetadata.AbsolutePath)), datasetID, finfo.Name, fileContent, iopts)
+
+	return err
+}
+
+func (c *StandaloneClient) IngestPaths(ctx context.Context, datasetID string, opts *IngestPathsOpts, paths ...string) (int, int, error) {
+	if strings.HasPrefix(paths[0], "ws://") {
+		if len(paths) > 1 {
+			return 0, 0, fmt.Errorf("cannot ingest multiple paths from workspace")
+		}
```

Comment:
I think this would not catch the case where the user passes a normal path as the first parameter, and a ws:// path as the second.

Reply:
But we're only checking the very first path for the ws:// prefix. If it's on the second item in the `paths` slice, it won't get caught.

Reply:
ws:// is overloaded, but fairly clear with regards to knowledge (which is file-based) I guess 🤷‍♂️

Anyway, those checks are "superficial", as only Otto should be using this stuff.

But you're right and I will fix the checks later on 👍

Reply:
i'm willing to accept these shortcomings (that it only checks the first path and that ws is overloaded).

@g-linville @StrongMonkey if this is your only issue with the PR, can your finish your review?